<a href="https://colab.research.google.com/github/asyahdin/Introduction-to-Python/blob/main/Exercises_DataWrangling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

mydata = pd.read_csv('NHANES.csv')
mydata.head(10)

SEQN  ALQ101  ALQ110  ALQ130  SMQ020  RIAGENDR  RIDAGEYR  RIDRETH1  \
0  83732     1.0     NaN     1.0       1         1        62         3   
1  83733     1.0     NaN     6.0       1         1        53         3   
2  83734     1.0     NaN     NaN       1         1        78         3   
3  83735     2.0     1.0     1.0       2         2        56         3   
4  83736     2.0     1.0     1.0       2         2        42         4   
5  83737     2.0     2.0     NaN       2         2        72         1   
6  83741     1.0     NaN     8.0       1         1        22         4   
7  83742     1.0     NaN     1.0       2         2        32         1   
8  83743     NaN     NaN     NaN       2         1        18         5   
9  83744     1.0     NaN     NaN       2         1        56         4   

   DMDCITZN  DMDEDUC2  ...  BPXSY2  BPXDI2  BMXWT  BMXHT  BMXBMI  BMXLEG  \
0       1.0       5.0  ...   124.0    64.0   94.8  184.5    27.8    43.3   
1       2.0       3.0  ...   140.0    88.0   90.4  171.4    30.8    38.0   
2       1.0       3.0  ...   132.0    44.0   83.4  170.1    28.8    35.6   
3       1.0       5.0  ...   134.0    68.0  109.8  160.9    42.4    38.5   
4       1.0       4.0  ...   114.0    54.0   55.2  164.9    20.3    37.4   
5       2.0       2.0  ...   122.0    58.0   64.4  150.0    28.6    34.4   
6       1.0       4.0  ...   112.0    74.0   76.6  165.4    28.0    38.8   
7       2.0       4.0  ...   114.0    70.0   64.5  151.3    28.2    34.1   
8       1.0       NaN  ...     NaN     NaN   72.4  166.1    26.2     NaN   
9       1.0       3.0  ...   180.0   114.0  108.3  179.4    33.6    46.0   

   BMXARML  BMXARMC  BMXWAIST  HIQ210  
0     43.6     35.9     101.1     2.0  
1     40.0     33.2     107.9     NaN  
2     37.0     31.0     116.5     2.0  
3     37.7     38.3     110.1     2.0  
4     36.0     27.2      80.4     2.0  
5     33.5     31.4      92.9     NaN  
6     38.0     34.0      86.6     NaN  
7     33.1     31.5      93.3     2.0  
8      NaN      NaN       NaN     2.0  
9     44.1     38.5     116.0     2.0  

[10 rows x 28 columns]

In [ ]:
mydata.shape

(5735, 28)

In [ ]:
mydata.columns #columns name

In [ ]:
mydata.info() #a data from dataset #want to know which is missing value in column

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5735 entries, 0 to 5734
Data columns (total 28 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEQN      5735 non-null   int64  
 1   ALQ101    5208 non-null   float64
 2   ALQ110    1731 non-null   float64
 3   ALQ130    3379 non-null   float64
 4   SMQ020    5735 non-null   int64  
 5   RIAGENDR  5735 non-null   int64  
 6   RIDAGEYR  5735 non-null   int64  
 7   RIDRETH1  5735 non-null   int64  
 8   DMDCITZN  5734 non-null   float64
 9   DMDEDUC2  5474 non-null   float64
 10  DMDMARTL  5474 non-null   float64
 11  DMDHHSIZ  5735 non-null   int64  
 12  WTINT2YR  5735 non-null   float64
 13  SDMVPSU   5735 non-null   int64  
 14  SDMVSTRA  5735 non-null   int64  
 15  INDFMPIR  5134 non-null   float64
 16  BPXSY1    5401 non-null   float64
 17  BPXDI1    5401 non-null   float64
 18  BPXSY2    5535 non-null   float64
 19  BPXDI2    5535 non-null   float64
 20  BMXWT     5666 non-null   floa

In [ ]:
mydata.describe(include='all') #describe all columns

SEQN       ALQ101       ALQ110       ALQ130       SMQ020  \
count   5735.000000  5208.000000  1731.000000  3379.000000  5735.000000   
mean   88678.583435     1.336406     1.587522     3.911512     1.607149   
std     2882.139237     0.505514     0.623940    34.341839     0.571975   
min    83732.000000     1.000000     1.000000     1.000000     1.000000   
25%    86164.000000     1.000000     1.000000     1.000000     1.000000   
50%    88668.000000     1.000000     2.000000     2.000000     2.000000   
75%    91178.500000     2.000000     2.000000     3.000000     2.000000   
max    93702.000000     9.000000     9.000000   999.000000     9.000000   

          RIAGENDR     RIDAGEYR     RIDRETH1     DMDCITZN     DMDEDUC2  ...  \
count  5735.000000  5735.000000  5735.000000  5734.000000  5474.000000  ...   
mean      1.518919    48.052310     3.042371     1.185385     3.441725  ...   
std       0.499686    18.431011     1.296793     0.491678     1.309700  ...   
min       1.000000    18.000000     1.000000     1.000000     1.000000  ...   
25%       1.000000    32.000000     2.000000     1.000000     3.000000  ...   
50%       2.000000    48.000000     3.000000     1.000000     4.000000  ...   
75%       2.000000    63.000000     4.000000     1.000000     4.750000  ...   
max       2.000000    80.000000     5.000000     9.000000     9.000000  ...   

            BPXSY2       BPXDI2        BMXWT        BMXHT       BMXBMI  \
count  5535.000000  5535.000000  5666.000000  5673.000000  5662.000000   
mean    124.783017    69.346703    81.342676   166.142834    29.382197   
std      18.527012    13.022829    21.764409    10.079264     7.095921   
min      84.000000     0.000000    32.400000   129.700000    14.500000   
25%     112.000000    62.000000    65.900000   158.700000    24.300000   
50%     122.000000    70.000000    78.200000   166.000000    28.300000   
75%     134.000000    78.000000    92.700000   173.500000    33.000000   
max     238.000000   144.000000   198.900000   202.700000    67.300000   

            BMXLEG      BMXARML      BMXARMC     BMXWAIST       HIQ210  
count  5345.000000  5427.000000  5427.000000  5368.000000  4732.000000  
mean     38.576782    37.146987    33.112235    99.567213     1.915469  
std       3.873018     2.800784     5.268027    16.844109     0.414845  
min      26.000000    28.200000    17.100000    58.700000     1.000000  
25%      36.000000    35.200000    29.500000    87.600000     2.000000  
50%      38.600000    37.100000    32.700000    98.300000     2.000000  
75%      41.200000    39.000000    36.200000   109.300000     2.000000  
max      51.500000    47.400000    58.400000   171.600000     9.000000  

[8 rows x 28 columns]

In [ ]:
pd.set_option('display.max_columns', None)

mydata.head(10)

SEQN  ALQ101  ALQ110  ALQ130  SMQ020  RIAGENDR  RIDAGEYR  RIDRETH1  \
0  83732     1.0     NaN     1.0       1         1        62         3   
1  83733     1.0     NaN     6.0       1         1        53         3   
2  83734     1.0     NaN     NaN       1         1        78         3   
3  83735     2.0     1.0     1.0       2         2        56         3   
4  83736     2.0     1.0     1.0       2         2        42         4   
5  83737     2.0     2.0     NaN       2         2        72         1   
6  83741     1.0     NaN     8.0       1         1        22         4   
7  83742     1.0     NaN     1.0       2         2        32         1   
8  83743     NaN     NaN     NaN       2         1        18         5   
9  83744     1.0     NaN     NaN       2         1        56         4   

   DMDCITZN  DMDEDUC2  DMDMARTL  DMDHHSIZ   WTINT2YR  SDMVPSU  SDMVSTRA  \
0       1.0       5.0       1.0         2  134671.37        1       125   
1       2.0       3.0       3.0         1   24328.56        1       125   
2       1.0       3.0       1.0         2   12400.01        1       131   
3       1.0       5.0       6.0         1  102718.00        1       131   
4       1.0       4.0       3.0         5   17627.67        2       126   
5       2.0       2.0       4.0         5   11252.31        1       128   
6       1.0       4.0       5.0         3   37043.09        2       128   
7       2.0       4.0       1.0         4   22744.36        1       125   
8       1.0       NaN       NaN         3   18526.16        2       122   
9       1.0       3.0       3.0         1   20395.54        2       126   

   INDFMPIR  BPXSY1  BPXDI1  BPXSY2  BPXDI2  BMXWT  BMXHT  BMXBMI  BMXLEG  \
0      4.39   128.0    70.0   124.0    64.0   94.8  184.5    27.8    43.3   
1      1.32   146.0    88.0   140.0    88.0   90.4  171.4    30.8    38.0   
2      1.51   138.0    46.0   132.0    44.0   83.4  170.1    28.8    35.6   
3      5.00   132.0    72.0   134.0    68.0  109.8  160.9    42.4    38.5   
4      1.23   100.0    70.0   114.0    54.0   55.2  164.9    20.3    37.4   
5      2.82   116.0    58.0   122.0    58.0   64.4  150.0    28.6    34.4   
6      2.08   110.0    70.0   112.0    74.0   76.6  165.4    28.0    38.8   
7      1.03   120.0    70.0   114.0    70.0   64.5  151.3    28.2    34.1   
8      5.00     NaN     NaN     NaN     NaN   72.4  166.1    26.2     NaN   
9      1.19   178.0   116.0   180.0   114.0  108.3  179.4    33.6    46.0   

   BMXARML  BMXARMC  BMXWAIST  HIQ210  
0     43.6     35.9     101.1     2.0  
1     40.0     33.2     107.9     NaN  
2     37.0     31.0     116.5     2.0  
3     37.7     38.3     110.1     2.0  
4     36.0     27.2      80.4     2.0  
5     33.5     31.4      92.9     NaN  
6     38.0     34.0      86.6     NaN  
7     33.1     31.5      93.3     2.0  
8      NaN      NaN       NaN     2.0  
9     44.1     38.5     116.0     2.0

In [ ]:
mydata.isnull().sum() #want to know how manyy missing value each colunm

SEQN           0
ALQ101       527
ALQ110      4004
ALQ130      2356
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

*Column with missing value**


ALQ101   :    527

ALQ110   :   4004

ALQ130   :   2356

DMDCITZN   :    1

DMDEDUC2  :   261

DMDMARTL  :   261

INDFMPIR  :   601

BPXSY1    :   334

BPXDI1    :   334

BPXSY2    :   200

BPXDI2    :   200

BMXWT    :     69

BMXHT      :   62

BMXBMI    :    73

BMXLEG     :  390

BMXARML   :   308

BMXARMC   :   308

BMXWAIST   :  367

HIQ210   :   1003





In [ ]:
mean_alq101 = mydata['ALQ101'].astype('float').mean() #convert to float
mean_alq101 #got the value of gender

1.336405529953917

In [ ]:
import numpy as np

mydata['ALQ101'].replace(np.nan, mean_alq101, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110      4004
ALQ130      2356
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_alq110 = mydata['ALQ110'].astype('float').mean() #convert to float
mean_alq110

1.5875216637781628

In [ ]:
mydata['ALQ110'].replace(np.nan, mean_alq110, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130      2356
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_alq130 = mydata['ALQ130'].astype('float').mean() #convert to float
mean_alq130

3.9115122817401597

In [ ]:
mydata['ALQ130'].replace(np.nan, mean_alq130, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130         0
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_dmdcitzn = mydata['DMDCITZN'].astype('float').mean()
mydata['DMDCITZN'].replace(np.nan, mean_dmdcitzn, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130         0
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       0
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_dmdeduc2 = mydata['DMDEDUC2'].astype('float').mean()
mydata['DMDEDUC2'].replace(np.nan, mean_dmdeduc2, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130         0
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       0
DMDEDUC2       0
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_dmdmartl = mydata['DMDMARTL'].astype('float').mean()
mydata['DMDMARTL'].replace(np.nan, mean_dmdmartl, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130         0
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       0
DMDEDUC2       0
DMDMARTL       0
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_indfmpir = mydata['INDFMPIR'].astype('float').mean()
mydata['INDFMPIR'].replace(np.nan, mean_indfmpir, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130         0
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       0
DMDEDUC2       0
DMDMARTL       0
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR       0
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_bpxsy1 = mydata['BPXSY1'].astype('float').mean()
mydata['BPXSY1'].replace(np.nan, mean_bpxsy1, inplace=True)
mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130         0
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       0
DMDEDUC2       0
DMDMARTL       0
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR       0
BPXSY1         0
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mean_bpxdi1 = mydata['BPXDI1'].astype('float').mean()
mydata['BPXDI1'].replace(np.nan, mean_bpxdi1, inplace=True)

mean_bpxsy2 = mydata['BPXSY2'].astype('float').mean()
mydata['BPXSY2'].replace(np.nan, mean_bpxsy2, inplace=True)

mean_bpxdi2 = mydata['BPXDI2'].astype('float').mean()
mydata['BPXDI2'].replace(np.nan, mean_bpxdi2, inplace=True)

mean_bmxwt = mydata['BMXWT'].astype('float').mean()
mydata['BMXWT'].replace(np.nan, mean_bmxwt, inplace=True)

mean_bmxht = mydata['BMXHT'].astype('float').mean()
mydata['BMXHT'].replace(np.nan, mean_bmxht, inplace=True)

mean_bmxbmi = mydata['BMXBMI'].astype('float').mean()
mydata['BMXBMI'].replace(np.nan, mean_bmxbmi, inplace=True)

mean_bmxleg = mydata['BMXLEG'].astype('float').mean()
mydata['BMXLEG'].replace(np.nan, mean_bmxleg, inplace=True)

mean_bmxarml = mydata['BMXARML'].astype('float').mean()
mydata['BMXARML'].replace(np.nan, mean_bmxarml, inplace=True)

mean_bmxarmc = mydata['BMXARMC'].astype('float').mean()
mydata['BMXARMC'].replace(np.nan, mean_bmxarmc, inplace=True)

mean_bmxwaist = mydata['BMXWAIST'].astype('float').mean()
mydata['BMXWAIST'].replace(np.nan, mean_bmxwaist, inplace=True)

mean_hiq210 = mydata['HIQ210'].astype('float').mean()
mydata['HIQ210'].replace(np.nan, mean_, inplace=True)


mydata.isnull().sum()

SEQN        0
ALQ101      0
ALQ110      0
ALQ130      0
SMQ020      0
RIAGENDR    0
RIDAGEYR    0
RIDRETH1    0
DMDCITZN    0
DMDEDUC2    0
DMDMARTL    0
DMDHHSIZ    0
WTINT2YR    0
SDMVPSU     0
SDMVSTRA    0
INDFMPIR    0
BPXSY1      0
BPXDI1      0
BPXSY2      0
BPXDI2      0
BMXWT       0
BMXHT       0
BMXBMI      0
BMXLEG      0
BMXARML     0
BMXARMC     0
BMXWAIST    0
HIQ210      0
dtype: int64

**DATA FORMATING**

In [ ]:
mydata.dtypes

SEQN          int64
ALQ101      float64
ALQ110      float64
ALQ130      float64
SMQ020        int64
RIAGENDR      int64
RIDAGEYR      int64
RIDRETH1      int64
DMDCITZN    float64
DMDEDUC2    float64
DMDMARTL    float64
DMDHHSIZ      int64
WTINT2YR    float64
SDMVPSU       int64
SDMVSTRA      int64
INDFMPIR    float64
BPXSY1      float64
BPXDI1      float64
BPXSY2      float64
BPXDI2      float64
BMXWT       float64
BMXHT       float64
BMXBMI      float64
BMXLEG      float64
BMXARML     float64
BMXARMC     float64
BMXWAIST    float64
HIQ210      float64
dtype: object

In [ ]:
mydata.describe(include='all') #describe all columns

SEQN       ALQ101       ALQ110       ALQ130       SMQ020  \
count   5735.000000  5735.000000  5735.000000  5735.000000  5735.000000   
mean   88678.583435     1.336406     1.587522     3.911512     1.607149   
std     2882.139237     0.481723     0.342718    26.358733     0.571975   
min    83732.000000     1.000000     1.000000     1.000000     1.000000   
25%    86164.000000     1.000000     1.587522     2.000000     1.000000   
50%    88668.000000     1.000000     1.587522     3.911512     2.000000   
75%    91178.500000     2.000000     1.587522     3.911512     2.000000   
max    93702.000000     9.000000     9.000000   999.000000     9.000000   

          RIAGENDR     RIDAGEYR     RIDRETH1     DMDCITZN     DMDEDUC2  \
count  5735.000000  5735.000000  5735.000000  5735.000000  5735.000000   
mean      1.518919    48.052310     3.042371     1.185385     3.441725   
std       0.499686    18.431011     1.296793     0.491635     1.279545   
min       1.000000    18.000000     1.000000     1.000000     1.000000   
25%       1.000000    32.000000     2.000000     1.000000     3.000000   
50%       2.000000    48.000000     3.000000     1.000000     4.000000   
75%       2.000000    63.000000     4.000000     1.000000     4.000000   
max       2.000000    80.000000     5.000000     9.000000     9.000000   

          DMDMARTL     DMDHHSIZ       WTINT2YR      SDMVPSU     SDMVSTRA  \
count  5735.000000  5735.000000    5735.000000  5735.000000  5735.000000   
mean      2.628608     3.323801   40312.412352     1.487881   126.236617   
std       2.312293     1.724670   38768.921774     0.499897     4.244406   
min       1.000000     1.000000    5330.960000     1.000000   119.000000   
25%       1.000000     2.000000   17164.085000     1.000000   123.000000   
50%       2.000000     3.000000   24654.860000     1.000000   126.000000   
75%       5.000000     5.000000   42862.305000     2.000000   130.000000   
max      77.000000     7.000000  233755.840000     2.000000   133.000000   

          INDFMPIR       BPXSY1       BPXDI1       BPXSY2       BPXDI2  \
count  5735.000000  5735.000000  5735.000000  5735.000000  5735.000000   
mean      2.403204   125.084614    69.516386   124.783017    69.346703   
std       1.515717    17.934551    12.500776    18.201036    12.793698   
min       0.000000    82.000000     0.000000    84.000000     0.000000   
25%       1.165000   114.000000    62.000000   112.000000    62.000000   
50%       2.260000   124.000000    69.516386   124.000000    69.346703   
75%       3.450000   134.000000    76.000000   134.000000    76.000000   
max       5.000000   236.000000   120.000000   238.000000   144.000000   

             BMXWT        BMXHT       BMXBMI       BMXLEG      BMXARML  \
count  5735.000000  5735.000000  5735.000000  5735.000000  5735.000000   
mean     81.342676   166.142834    29.382197    38.576782    37.146987   
std      21.633062    10.024624     7.050607     3.738986     2.724524   
min      32.400000   129.700000    14.500000    26.000000    28.200000   
25%      66.100000   158.750000    24.400000    36.100000    35.300000   
50%      78.400000   166.142834    28.400000    38.576782    37.146987   
75%      92.600000   173.400000    33.000000    41.000000    39.000000   
max     198.900000   202.700000    67.300000    51.500000    47.400000   

           BMXARMC     BMXWAIST       HIQ210  
count  5735.000000  5735.000000  5735.000000  
mean     33.112235    99.567213     1.915469  
std       5.124589    16.296149     0.376820  
min      17.100000    58.700000     1.000000  
25%      29.700000    88.400000     1.915469  
50%      33.000000    99.567213     2.000000  
75%      36.000000   108.400000     2.000000  
max      58.400000   171.600000     9.000000

**Data Normalization**
(Min - Max)

* ALQ130*
* DMDMARTL
* WTINT2YR*
* SDMVSTRA
* BPXSY1*
* BPXDI1
* BPXSY2
* BPX

USING MIN AND MAX

In [ ]:
mydata['ALQ130'] = (mydata['ALQ130'] - mydata['ALQ130'].min()) / (mydata['ALQ130'].max() - mydata['ALQ130'].min())

mydata[['ALQ130', 'WTINT2YR', 'BPXSY1']].head()

ALQ130   WTINT2YR  BPXSY1
0  0.000000  134671.37   128.0
1  0.005010   24328.56   146.0
2  0.002917   12400.01   138.0
3  0.000000  102718.00   132.0
4  0.000000   17627.67   100.0

In [ ]:
mydata['WTINT2YR'] = (mydata['WTINT2YR'] - mydata['WTINT2YR'].min()) / (mydata['WTINT2YR'].max() - mydata['WTINT2YR'].min())

mydata[['ALQ130', 'WTINT2YR', 'BPXSY1']].head()

ALQ130  WTINT2YR  BPXSY1
0  0.000000  0.566227   128.0
1  0.005010  0.083168   146.0
2  0.002917  0.030947   138.0
3  0.000000  0.426342   132.0
4  0.000000  0.053833   100.0

In [ ]:
mydata['BPXSY1'] = (mydata['BPXSY1'] - mydata['BPXSY1'].min()) / (mydata['BPXSY1'].max() - mydata['BPXSY1'].min())

mydata[['ALQ130', 'WTINT2YR', 'BPXSY1']].head()

ALQ130  WTINT2YR    BPXSY1
0  0.000000  0.566227  0.298701
1  0.005010  0.083168  0.415584
2  0.002917  0.030947  0.363636
3  0.000000  0.426342  0.324675
4  0.000000  0.053833  0.116883

In [ ]:
mydata[['ALQ130', 'WTINT2YR', 'BPXSY1']].describe()

ALQ130     WTINT2YR       BPXSY1
count  5735.000000  5735.000000  5735.000000
mean      0.002917     0.153142     0.279770
std       0.026412     0.169723     0.116458
min       0.000000     0.000000     0.000000
25%       0.001002     0.051803     0.207792
50%       0.002917     0.084596     0.272727
75%       0.002917     0.164305     0.337662
max       1.000000     1.000000     1.000000